driver-core/src/main/com/mongodb/operation/CommandOperationHelper.java

```diff
@@ -554,6 +557,7 @@ private void checkRetryableException(final Throwable originalError, final Single
             }
 
             private void retryableCommand(final Throwable originalError) {
+                logRetryExecute(command.getFirstKey(), originalError);
```

Comment:
Can we call logUnableToRetry on line 551 of checkRetryableException?  Otherwise, the async logging is different than the sync logging.

Reply:
Done